src/atom.cpp

```diff
@@ -1884,11 +1884,19 @@ void Atom::setup_sort_bins()
   // user setting if explicitly set
   // default = 1/2 of neighbor cutoff
   // check if neighbor cutoff = 0.0
+  // and in that case, disable sorting
 
   double binsize;
   if (userbinsize > 0.0) binsize = userbinsize;
-  else binsize = 0.5 * neighbor->cutneighmax;
-  if (binsize == 0.0) error->all(FLERR,"Atom sorting has bin size = 0.0");
```

Comment:
@sjplimp this "Atom sorting has bin size = 0.0" error has always been bothering me; the fact that there is an error and particularly the somewhat cryptic message that comes with it. I think just disabling sorting (and printing a WARNING message) is the better behavior. Without a pair style providing a cutoff and thus a default bin size, there is little use for atom sorting in the first place. I just had another e-mail in my private inbox by another confused LAMMPS user, so I decided to do something about it.